src/include/libminiasync/future.h

```diff
@@ -63,7 +63,12 @@
 
 #include <stddef.h>
 #include <stdint.h>
+#include "core/util.h"
```

Comment:
We can't include this file here since future.h is a public header. The main reason we added pause in the busy poll macro is to let valgrind schedule threads - before this valgrind was stuck in the empty loop.

Reply:
So I think we need to move all ifdefs into the header, and maybe use `sched_yield()` on non-x86 platforms. I know it's an anti-pattern but I don't have a better idea right now :P

Reply:
@pbalcer Oh! Sorry, I had missed this.
I removed the dependency on internal headers by adding a macro that uses `sched_yield()` as you suggested.
But I'm not sure if this is matching your proposal.  Could you review it again, please?

Reply:
Yes, that's what I meant. Thanks!